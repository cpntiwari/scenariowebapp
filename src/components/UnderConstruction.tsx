import React from "react";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import { NavLink } from "react-router-dom";
import { Button } from "@material-ui/core";
import { Palette } from "../themes/Palette";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    textAlign: "center",
    color: Palette.colors.green["500"]
  },
  button: {
    color: Palette.colors.blue["700"],
    textDecoration: "none",
    cursor: "pointer"
  },
  main: {
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2),
  },
}));

export default function ScenarioDetails() {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Container component="main" className={classes.main} maxWidth="sm">
        <Typography variant="h2" component="h1" gutterBottom>
          Coming Soon
        </Typography>
        <Typography variant="h5" component="h2" gutterBottom>
          Our page is in under contruction.
        </Typography>
        <Typography variant="body1">
          <NavLink className={classes.button} to="/">
            <Button variant="outlined" color="primary"> Go back to Scenario Manager</Button>
          </NavLink>
        </Typography>
      </Container>
    </div>
  );
}
