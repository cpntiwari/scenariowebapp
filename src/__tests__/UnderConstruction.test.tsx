import React from "react";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import UnderConstruction from "../components/UnderConstruction";
import { shallow } from "enzyme";
let wrapper: any;

beforeEach(() => {
  wrapper = shallow(<UnderConstruction />);
});

describe("Under Construction Component Rendering", () => {
  test("should test Under construction component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should render a <Container /> to display the maximum length ", () => {
    expect(wrapper.find(Container)).toHaveLength(1);
  });

  test("should render a <Typography /> to display the maximum length of the comment", () => {
    expect(wrapper.find(Typography)).toHaveLength(3);
  });
});
