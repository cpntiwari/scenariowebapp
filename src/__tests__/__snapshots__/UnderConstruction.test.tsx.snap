// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Under Construction Component Rendering should test Under construction component 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-main-2"
    component="main"
    maxWidth="sm"
  >
    <WithStyles(ForwardRef(Typography))
      component="h1"
      gutterBottom={true}
      variant="h2"
    >
      Coming Soon
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      Our page is under contruction.
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      <NavLink
        style={
          Object {
            "textDecoration": "none",
          }
        }
        to="/"
      >
        Go to the Scenario Manager
      </NavLink>
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
